pkg/kubelet/container/runtime_cache.go

```diff
@@ -80,6 +80,14 @@ func (r *runtimeCache) GetPods() ([]*Pod, error) {
 	return r.pods, nil
 }
 
+// getPods returns the current cached pods. This is used mainly for testing
+// because it does not start the refreshing thread.
+func (r *runtimeCache) getCachedPods() []*Pod {
+	r.Lock()
+	defer r.Unlock()
```

Comment:
how do these locks help? it's a little confusing since you're anyway returning a list of pointers, are you trying to get this list at an exact time?